torchvision/prototype/datasets/_builtin/cityscapes.py

```diff
@@ -0,0 +1,211 @@
+import json
+from collections import namedtuple
+from functools import partial
+from pathlib import Path
+from typing import Any, Dict, List, Tuple
+
+from torchdata.datapipes.iter import Grouper, IterDataPipe, Mapper, Filter, IterKeyZipper
+from torchvision.prototype.datasets.utils import (
+    Dataset,
+    DatasetInfo,
+    DatasetConfig,
+    ManualDownloadResource,
+    OnlineResource,
+)
+from torchvision.prototype.datasets.utils._internal import INFINITE_BUFFER_SIZE, hint_sharding, hint_shuffling
+from torchvision.prototype.features import EncodedImage
+from torchvision.prototype.utils._internal import FrozenMapping
+
+
+class CityscapesDatasetInfo(DatasetInfo):
+    def __init__(self, *args: Any, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+        self._configs = tuple(
+            config
+            for config in self._configs
+            if not (
+                (config.split == "test" and config.mode == "coarse")
+                or (config.split == "train_extra" and config.mode == "fine")
+            )
+        )
+
+    def make_config(self, **options: Any) -> DatasetConfig:
+        config = super().make_config(**options)
+        if config.split == "test" and config.mode == "coarse":
+            raise ValueError("`split='test'` is only available for `mode='fine'`")
+        if config.split == "train_extra" and config.mode == "fine":
+            raise ValueError("`split='train_extra'` is only available for `mode='coarse'`")
+
+        return config
+
+
+class CityscapesResource(ManualDownloadResource):
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(
+            "Register on https://www.cityscapes-dataset.com/login/ and follow the instructions there.", **kwargs
+        )
+
+
+CityscapesClass = namedtuple(
+    "CityscapesClass",
+    ["name", "id", "train_id", "category", "category_id", "has_instances", "ignore_in_eval", "color"],
+)
+
+
+class Cityscapes(Dataset):
+
+    categories_to_details: FrozenMapping = FrozenMapping(
+        {
+            "unlabeled": CityscapesClass("unlabeled", 0, 255, "void", 0, False, True, (0, 0, 0)),
```

Comment:
It seems, all of this information is exposed to the user, but never used internally. Is this just static information or would it make sense to include it in the samples?

Reply:
It is used in `extra=dict(classname_to_details=self.categories_to_details),` and is exposed to the user as in the previously implementation

Reply:
My point is: what does the user do with this information? I reckon `color` is useful to parse a segmentation mask. The `category` looks like what we called `super_category` in COCO. So maybe we can just provide two dictionaries in the extra information that map the category to its color and super category respectively?

Reply:
CityscapesClass provides more info that just color, category name, super category. It may happen that users could use other fields for visualization purposes and thus new dataset would miss that info.

Reply:
In the stable API this is exposed through the *undocumented* `torchvision.datasets.Cityscapes(...).classes` attribute. Given that the new API is BC breaking anyway, IMO we should aim to reduce the baggage from the old implementation. I indeed see use cases for returning a color to category mapping (maybe we should have this on the `SegmentationMask` feature) as well as the category to super category mapping. For the rest I would remove them for now and only put them back if someone requests it. cc @NicolasHug for an opinion.

Reply:
It is dataset meta-data, probably coming from https://github.com/mcordts/cityscapesScripts and is used for visualization (this can be seen by searching through the GitHub and exploring repository by repository, https://github.com/search?p=1&q=CityscapesClass&type=Code). Even if this wasn't explicitly documented previously, I do not think that we have to remove that in the new API.